src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -35,6 +35,24 @@ bool goto_convertt::has_function_call(const exprt &expr)
   return false;
 }
 
+bool goto_convertt::has_sideeffect(const exprt &expr)
+{
+  // TODO: In the future, we want to allow some function calls,
+  // specifically those which have an empty ASSIGNS clause annotation.
+  if(expr.id() == ID_side_effect)
+  {
+    return true;
+  }
+
+  for(const auto &op : expr.operands())
+  {
+    if(goto_convertt::has_sideeffect(op))
+      return true;
+  }
+
+  return false;
+}
+
```

Comment:
Good call out that this was never implemented. That said, can you actually do this a bit different as follows:
1) Get rid of the declaration as well.
2) At the point where you will currently need it, just use `has_subexpr(expr, ID_side_effect);` (from `util/expr_util.h`).